src/seedu/addressbook/commands/Command.java

```diff
@@ -54,6 +54,13 @@ public void setData(AddressBook addressBook, List<? extends ReadOnlyPerson> rele
         this.relevantPersons = relevantPersons;
     }
 
+    /**
+     * Determines whether the command will mutate the data in address book.
+     */
+    public boolean isMutating() {
+        return false;
```

Comment:
Having a default implementation for the `isMutating` method may not be intuitive for a class like `Command` as it is abstract in context. Why return false and not true?

Perhaps a better implementation would be make this method abstract and force all classes extended from `Command` to implement their own method definition.